File: Code/BeforeLegends/Assets/Scripts/Victory and Death Conditions/GameOver.js
#pragma strict

var gameOver : UI.Text;
var gameObjectsToDeactivateOnGameOver : GameObject[];
@HideInInspector
var activePlayers : GameObject[];

function Update() {
	activePlayers = GameObject.FindGameObjectsWithTag("Player");
	if(activePlayers.Length == 0) {
		for(var gO : GameObject in gameObjectsToDeactivateOnGameOver) {
			gO.SetActive(false);
		}
		gameOver.gameObject.SetActive(true);
		Time.timeScale = 0;
	}
}

File: Code/BeforeLegends/Assets/Scripts/Victory and Death Conditions/GameOver.cs
using UnityEngine;
using UnityEngine.UI;
using System.Collections;

public class GameOver : MonoBehaviour {
	public Text gameOver;
	public GameObject[] gameObjectsToDeactivateOnGameOver;

	public GameObject[] activePlayers;

	// Use this for initialization
	void Start () {
	
	}
	
	// Update is called once per frame
	void Update () {
		activePlayers = GameObject.FindGameObjectsWithTag("Player");
		if(activePlayers.Length == 0) {
			foreach(GameObject gO in gameObjectsToDeactivateOnGameOver) {
				gO.SetActive(false);
			}
			gameOver.gameObject.SetActive(true);
			Time.timeScale = 0;
		}
	}
}